vignettes/TechnicalDocmentation.Rmd

```diff
@@ -0,0 +1,636 @@
+the corresponding observations; $s_l = 0$ otherwise.
+
+In our code, $\boldsymbol{s}$ correponds to `vZeroIndsnew`, `vZeroIndsold`, and the 
+number of non-zero elements in $\boldsymbol{s}$ denoted by `nzeros_new`, and `nzeros_old`.
+
+Updates to $\boldsymbol{s}$ are performed via 
+
+```{r,echo=TRUE,eval=FALSE}
+for(int l = 0; l < num_edges; l++){
+  Eigen::VectorXi v_index = IndMat.row(l);
+  if(extract(v_new, v_index).sum() == 0){
+    vZeroIndsnew(l) = 1;
+  }
+}
+```
+
+Note, changes in the sparsity pattern (`vZeroIndsnew`) are one way, updating
+to $1$ if a column of zeros is observed, and not resetting to $0$ is a 
+fission occurs. Hence fusions are recorded in the order they happen, and 
+fissions are not represented.
+
+The CARP algorithm is now modified to terminate once all fusions have been 
+observed.
+
+1. Input: $\boldsymbol{x},\in \mathbb{R}^{n p}$, $\boldsymbol{w} \in \mathbb{R}^{\binom{n}{2}}$, $\lambda^{(0)} = \epsilon > 0, t > 0$
+
+2. Initialize: $\boldsymbol{u}^{(0)} = \boldsymbol{x}$, $\boldsymbol{v}^{(0)}_l,\boldsymbol{z}^{(0)}_l = \boldsymbol{x}_{l_1} - \boldsymbol{x}_{l_2}$
+
+3. `while` $\| s \|^2_2 \neq |E|$
+
+    a. $\textrm{CARPSS}(\boldsymbol{x},\boldsymbol{u}^{(k)},\boldsymbol{v}^{(k)},\boldsymbol{z}^{(k)},\boldsymbol{w},\lambda^{(k)})$
+    
+    b. $\lambda^{(k+1)} = t\lambda^{(k)}$
+    
+    
+
+
+#### CBASS
+
+Our `CBASS` duplicates the `CARP` updates above based on the Dykstra-like 
+proximal algorithm  of Convex Biclustering.
+ 
+Recall the Convex Biclusering algorithm is given by
+
+1. Input: $\boldsymbol{X} \in \mathbb{R}^{p \times n}$, $\boldsymbol{w}_c \in \mathbb{R}^{\binom{n}{2}}$,
+$\boldsymbol{w}_r \in \mathbb{R}^{\binom{p}{2}}$, $\lambda > 0$
+
+2. Initialize: $\boldsymbol{U}^0 = \boldsymbol{X}$, $\boldsymbol{P}^0 = \boldsymbol{0} \in \mathbb{R}^{p \times n}$,
+$\boldsymbol{Q}^0 = \boldsymbol{0} \in \mathbb{R}^{p \times n}$
+
+    a. $(\boldsymbol{Y}^{k})^T = \textrm{ConvexClustering}( (\boldsymbol{U}^k)^T + (\boldsymbol{P}^k)^T, \boldsymbol{w}_r, \lambda)$
+    
+    b. $\boldsymbol{P}^{k+1} = \boldsymbol{U}^k + \boldsymbol{P}^k - \boldsymbol{Y}^k$
+    
+    c. $\boldsymbol{U}^{k+1} = \textrm{ConvexClustering}( \boldsymbol{Y}^{k} + \boldsymbol{Q}^{k},\boldsymbol{w}_c,\lambda)$
+    
+    d. $\boldsymbol{Q}^{k+1} = \boldsymbol{Y}^k + \boldsymbol{Q}^k - \boldsymbol{U}^{k+1}$
+    
+
+where here $\textrm{ConvexClustering}(\boldsymbol{X},\boldsymbol{w},\lambda)$ 
+performs Convex Clustering on the columns of $\boldsymbol{X}$, using the 
+weights $\boldsymbol{w}$ at regularization amount $\lambda > 0$, returning 
+a matrix the same dimensions as $\boldsymbol{X}$.
+
+
+`CBASS` replaces calls to $\textrm{ConvexClustering}$ with calls to `CARPSS`, 
+unvec-ing and performing transposes as necessary:
+
+
+The update (a)
+$$
+(\boldsymbol{Y}^{k})^T = \textrm{ConvexClustering}( (\boldsymbol{U}^k)^T + (\boldsymbol{P}^k)^T, \boldsymbol{w}_r, \lambda)
+$$
+becomes in `CBASS`
+
+
+```{r,eval=FALSE,echo=TRUE}
```

Comment:
This isn't `R` code: maybe use `knitr`'s `Rcpp` engine to get better formatting.